### PR TITLE
(refactor) Make the title and description optional for most error classes

### PR DESCRIPTION
**Pull request by stiefel40k**
The changes should cause no breaks, however as I ran the tests some of the tests failed, but the problem was with the Wrapper and some module imports

### CODE REVIEW

**Comment by stiefel40k** [comment]
Hi @kgriffs!

As you can see I modified the docstrings as you requested and I also created some simple tests for testing the new constructors. I hope now it satisfies the requirements.

Sorry for the multiple commits, and the build errors, but my IDE was configured sligthly in an other way, so the PEP8 checker caused the build error.

File: falcon/errors.py [hunk 1]
```diff
@@ -1174,12 +1147,12 @@ class HTTPServiceUnavailable(HTTPError):
...
     """
 
-    def __init__(self, title, description, retry_after, **kwargs):
+    def __init__(self, title=None, description=None, retry_after=None, **kwargs):
         headers = kwargs.setdefault('headers', {})
 
         if isinstance(retry_after, datetime):
             headers['Retry-After'] = util.dt_to_http(retry_after)
-        else:
+        elif retry_after is not None:
```
Comment (kgriffs) [inline falcon/errors.py]: Nice catch!

**Comment by kgriffs** [comment]
@stiefel40k would you mind rebasing on master just to make sure the gates pass before we merge?

**Comment by stiefel40k** [comment]
@kgriffs what do you mean rebasing? pulling the new changes from your master and resolve possible conflicts?

**Comment by kgriffs** [comment]
Oh, no worries. You have a couple of options. First, you can just use the "update branch" GitHub button below (it's disabled for me since I don't have push access to your fork). It's not quite the same as a rebase, but it works. Alternatively, assuming you have added an "upstream" remote to point to falconry/falcon, you can fetch that and then rebase your branch onto it; something like:

```bash
$ git fetch upstream master
$ git rebase upstream/master
```

**Pull request merged**